src/app/App.js

```diff
@@ -0,0 +1,207 @@
+
+    'app/config',
+    'app/FilterContainer',
+
+    'dijit/registry',
+    'dijit/_TemplatedMixin',
+    'dijit/_WidgetBase',
+    'dijit/_WidgetsInTemplateMixin',
+
+    'dojo/dom',
+    'dojo/dom-style',
+    'dojo/on',
+    'dojo/text!app/templates/App.html',
+    'dojo/_base/array',
+    'dojo/_base/declare',
+    'dojo/_base/lang',
+
+    'esri/dijit/Print',
+    'esri/geometry/Extent',
+    'esri/layers/FeatureLayer',
+
+    'ijit/widgets/layout/SideBarToggler',
+
+    'layer-selector'
+], function (
+    FindAddress,
+    MagicZoom,
+    BaseMap,
+
+    Sherlock,
+    WebAPI,
+
+    config,
+    FilterContainer,
+
+    registry,
+    _TemplatedMixin,
+    _WidgetBase,
+    _WidgetsInTemplateMixin,
+
+    dom,
+    domStyle,
+    on,
+    template,
+    array,
+    declare,
+    lang,
+
+    Print,
+    Extent,
+    FeatureLayer,
+
+    SideBarToggler,
+
+    BaseMapSelector
+) {
+    return declare([_WidgetBase, _TemplatedMixin, _WidgetsInTemplateMixin], {
+        // summary:
+        //      The main widget for the app
+
+        widgetsInTemplate: true,
+        templateString: template,
+        baseClass: 'app',
+
+        // childWidgets: Object[]
+        //      container for holding custom child widgets
+        childWidgets: null,
+
+        //facilitiesId
+        //  layer ID for health facality points
+        facilitiesId: 'Healthfacilities',
+        facilityPoints: null,
+        map: null,
+
+        constructor: function () {
+            // summary:
+            //      first function to fire after page loads
+            console.info('app.App::constructor', arguments);
+
+            config.app = this;
+            this.childWidgets = [];
+
+            this.inherited(arguments);
+        },
+        postCreate: function () {
+            // summary:
+            //      Fires when
+            console.log('app.App::postCreate', arguments);
+
+            // set version number
+            // this.version.innerHTML = config.version;
+
+            this.initMap();
+            //countiesProvider: Provider
+            //      Provider for sherlock and zoom to city or county
+            var countiesProvider = new WebAPI(
+                config.apiKey,
+                'SGID10.LOCATION.ZoomLocations',
+                'NAME',
+                {
+                    wkid: 3857
+                }
+            );
+
+            this.childWidgets.push(
+                new SideBarToggler({
+                    sidebar: this.sideBar,
+                    map: this.map,
+                    centerContainer: this.centerContainer
+                }, this.sidebarToggle),
+                new Sherlock({
+                    provider: countiesProvider,
+                    map: this.map,
+                    maxResultsToDisplay: 10,
+                    placeHolder: 'Enter location'
+                }, this.cityNode),
+                this.filter = new FilterContainer({}, this.facalityFilterNode)
```

Comment:
does this syntax work? I think you should define it outside the array push and then push the reference.